repository/src/main/java/org/apache/atlas/tasks/TaskManagement.java

```diff
@@ -205,6 +205,14 @@ public void deleteByGuids(List<String> guids) throws AtlasBaseException {
         }
     }
 
+    public void softDelete(String guid) throws AtlasBaseException{
```

Comment:
Change method name to deleteByGuid and add another param -> deleteType which can be HARD/SOFT.